File: Algorithm/Baekjoon/23년도/23_05_week_2/27972.cpp
#include <iostream>
using namespace std;

bool chk = false;

int N, mx, pre_num, stack = 0;

int main()
{
	cin >> N >> pre_num;
	int now_num;
	for (int i = 1; i < N; i++) {
		cin >> now_num;
		if (pre_num < now_num) {
			if (chk) {
				stack++;
			}
			else {
				chk = true;
				stack = 1;
			}
			if (stack > mx) {
				mx = stack;
			}
		}
		else if (pre_num > now_num) {
			if (chk) {
				chk = false;
				stack = 1;
			}
			else {
				stack++;
			}
			if (stack > mx) {
				mx = stack;
			}
		}
		pre_num = now_num;
	}
	cout << mx + 1;
	
	return 0;

}

File: Algorithm/Baekjoon/23년도/23_05_week_2/27972.py
N = int(input())
lst = list(map(int, input().split()))

mx = 0
now_num = 0
pre_num = lst[0]
flag = False
for n in lst:
    if n > pre_num:
        if flag:
            now_num += 1
        else:
            flag = True
            now_num = 1

    elif n < pre_num:
        if flag:
            flag = False
            now_num = 1
        else:
            now_num += 1

    mx = max(mx, now_num)

    pre_num = n

print(mx + 1)
